### ISSUE TITLE
Errors in ActiveJob test suite after a clean install of rails-dev-box - "undefined method 'adapter_is?' for #<QueuingTest..."

### ISSUE THREAD

**Issue by richiethomas** [rails/rails#42974]
I took the liberty of creating a PR to address the issue [here](https://github.com/rails/rails/pull/42973).

### Steps to reproduce
1. Follow the instructions [here](https://github.com/rails/rails-dev-box) to set up a new instance of `rails-dev-box`, run `vagrant up` and `vagrant ssh`, etc. (not sure if this is necessary or not; you may see the same behavior I observed from your current rails-dev-box setup regardless of how fresh it is).
2. From within your local clone of the Rails repo contained in your Vagrant VM, run `cd activejob && bin/test`.

### Expected behavior

No errors or failures should be returned.

### Actual behavior
I see the following results:

```
Error:
QueuingTest#test_should_supply_a_provider_job_id_when_available_for_delayed_jobs:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a012607f18>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:87:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:86

E

Error:
QueuingTest#test_should_supply_a_wrapped_class_name_to_Sidekiq:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a01261af28>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:31:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:30

E

Error:
QueuingTest#test_current_timezone_is_kept_while_running_perform_later:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a01262e7f8>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:110:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:109

S.SE

Error:
QueuingTest#test_should_run_job_with_higher_priority_first:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a01264b5b0>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:126:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:125

E

Error:
QueuingTest#test_should_run_job_with_higher_priority_first_in_Backburner:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a012663b38>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:138:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:137

E

Error:
QueuingTest#test_current_locale_is_kept_while_running_perform_later:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a01267b3a0>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:93:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:92

E

Error:
QueuingTest#test_should_access_provider_job_id_inside_Sidekiq_job:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a01268a968>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:41:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:40

E

Error:
QueuingTest#test_should_run_job_enqueued_in_the_future_at_the_specified_time:
NameError: uninitialized constant QueuingTest::TestJob
    /vagrant/rails/activejob/test/integration/queuing_test.rb:71:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:70

E

Error:
QueuingTest#test_should_supply_a_provider_job_id_when_available_for_immediate_jobs:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a01149a2d0>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:81:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:80

.E

Error:
QueuingTest#test_should_not_run_job_enqueued_in_the_future:
NameError: uninitialized constant QueuingTest::TestJob
    /vagrant/rails/activejob/test/integration/queuing_test.rb:63:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:62

E

Error:
QueuingTest#test_should_run_jobs_enqueued_on_a_listening_queue:
NameError: uninitialized constant QueuingTest::TestJob
    /vagrant/rails/activejob/test/integration/queuing_test.rb:11:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:10

.E

Error:
QueuingTest#test_should_not_run_jobs_queued_on_a_non-listening_queue:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a011146e30>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:17:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:16

E

Error:
QueuingTest#test_should_supply_a_wrapped_class_name_to_DelayedJob:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a0111d42f8>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:49:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:48

E

Error:
QueuingTest#test_resque_JobWrapper_should_have_instance_variable_queue:
NoMethodError: undefined method `adapter_is?' for #<QueuingTest:0x000055a010f34a98>
    /vagrant/rails/activejob/test/integration/queuing_test.rb:56:in `block in <class:QueuingTest>'


bin/test test/integration/queuing_test.rb:55

..

Finished in 0.412302s, 916.8031 runs/s, 1903.9428 assertions/s.
378 runs, 785 assertions, 0 failures, 14 errors, 27 skips
```

### System configuration
**Rails version**:

From the `Tags` list in my fork of the `Rails` repo:
```
v6.1.4 release
 on Jun 24  8321702  zip  tar.gz
```

**Ruby version**:
Version on my machine- ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin19]
Version inside Vagrant- ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-linux-gnu]

**Comment by yahonda** [comment]
As far as I know, there are two ways to run unit tests of Rails modules like Active Record, Active Support .etc.
`bundle exec rake test` and `bin/test`. Rails CI at Buildkite runs them by `bundle exec rake test`.

In addition to unit tests, Active Job has integration tests with adapters which can be executed by `bundle exec rake test_integration`, which runs fine.
I prefer `bin/test` only runs unit tests and those who want to run Active Job integration tests can execute `bundle exec rake test_integration`.

I have pushed one commit to exclude integration tests via `bin/test`, which works fine as follows.
https://github.com/yahonda/rails/commit/3084df3a26e80030b65fad81a6e9392cda650df2


```ruby
$ cd activejob
$ bin/test
Using inline
Run options: --seed 44630

# Running:

......................................S.S...................SSSSSSSSSSSSSSSSSSSSSSSSS........................................................................................................................................................................................................................................................................................

Finished in 0.312099s, 1169.5021 runs/s, 2518.4347 assertions/s.
365 runs, 786 assertions, 0 failures, 0 errors, 27 skips

You have skipped tests. Run with --verbose for details.
$
```

**Comment by richiethomas** [comment]
@yahonda, a few beginner questions:

1. I observed the same passing tests that you reported in your comment, and I'm curious what could cause these specs to fail if we run `bin/test` vs. `bundle exec rake test integration`.  Are there certain setup stakes that the latter task executes, which the former task doesn't?  I tried answering this question myself by looking for the code that's executed when we enter the `bundle exec rake test integration` command, but I was unsuccessful in finding it.

2. Also, are there any conventions or expectations among the core team around whether `bin/test` only executes unit tests vs. unit + integration tests?   And would the suggested change abide by any such conventions?

3. Lastly, if we exclude the integration tests from the `bin/test` command, is it worth also updating the documentation to make it clear which specs are or aren't run, and how to run the excluded specs if desired?  If so, I'd be happy to attempt that update.
